caddyhttp/fastcgi/fastcgi.go

```diff
@@ -73,9 +82,30 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 
 			// Connect to FastCGI gateway
 			network, address := rule.parseAddress()
-			fcgiBackend, err := Dial(network, address)
-			if err != nil {
-				return http.StatusBadGateway, err
+
+			var fcgiBackend *FCGIClient
+			reuse := false
+			if usePersistentFcgiConnections {
+				metaMutex.Lock()
+				// re use connection, if possible
+				if _, ok := poorMansSerialisation[network+address]; !ok {
+					poorMansSerialisation[network+address] = new(sync.Mutex)
+				}
+				metaMutex.Unlock()
+
+				poorMansSerialisation[network+address].Lock()
+				defer poorMansSerialisation[network+address].Unlock()
```

Comment:
You are saying that I need to protect locking/unlocking the mutex because it's stored in a map?
If that's the case my map construction is pretty useless. I will change it.